pkgs/applications/video/epgstation/default.nix

```diff
@@ -1,85 +1,91 @@
 { lib
-, stdenv
+, fetchpatch
 , fetchFromGitHub
+, buildNpmPackage
 , makeWrapper
 , bash
 , nodejs
 , gzip
-, callPackage
+, python3
 }:
 
-let
-  # NOTE: use updateScript to bump the package version
-  pname = "EPGStation";
+buildNpmPackage rec {
+  pname = "epgstation";
   version = "2.6.20";
+
   src = fetchFromGitHub {
     owner = "l3tnun";
     repo = "EPGStation";
     rev = "v${version}";
     sha256 = "K1cAvmqWEfS6EY4MKAtjXb388XLYHtouxNM70PWgFig=";
   };
 
-  client = nodejs.pkgs.epgstation-client.override (drv: {
-    # This is set to false to keep devDependencies at build time. Build time
-    # dependencies are pruned afterwards.
-    production = false;
+  patches = [
+    ./use-mysql-over-domain-socket.patch
 
-    meta = drv.meta // {
-      inherit (nodejs.meta) platforms;
-    };
-  });
+    # upgrade dependencies to make it compatible with node 18
+    (fetchpatch {
+      url = "https://github.com/midchildan/EPGStation/commit/5d6cad7.patch";
+      sha256 = "sha256-9a8VUjczlyQHVO7w9MYorPIZunAuBuif1HNmtp1yMk8=";
+    })
+    (fetchpatch {
+      url = "https://github.com/midchildan/EPGStation/commit/c948e83.patch";
+      sha256 = "sha256-nM6KkVRURuQFZLXZ2etLU1a1+BoaJnfjngo07TFbe58=";
+    })
+  ];
 
-  server = nodejs.pkgs.epgstation.override (drv: {
-    # NOTE: updateScript relies on version matching the src.
-    inherit version src;
+  npmDepsHash = "sha256-dohencRGuvc+vSoclLVn5iles4GOuTq26BrEVeJ4GC4=";
+  npmBuildScript = "build-server";
+  npmRootPath = "/lib/node_modules/epgstation";
 
-    # This is set to false to keep devDependencies at build time. Build time
-    # dependencies are pruned afterwards.
-    production = false;
+  buildInputs = [ bash ];
+  nativeBuildInputs = [ makeWrapper python3 ];
 
-    buildInputs = (drv.buildInputs or [ ]) ++ [ bash ];
-    nativeBuildInputs = (drv.nativeBuildInputs or [ ]) ++ [
-      makeWrapper
-    ];
+  clientDir = buildNpmPackage {
+    pname = "${pname}-client";
+    inherit version src installPhase meta;
 
-    preRebuild = ''
-      # Fix for OpenSSL compat with newer Node.js
-      export NODE_OPTIONS=--openssl-legacy-provider
+    npmDepsHash = "sha256-a/cDPABWI4lPxvSOI4D90O71A9lm8icPMak/g6DPYQY=";
+    npmRootPath = "";
 
-      # Fix for not being able to connect to mysql using domain sockets.
-      patch -p1 < ${./use-mysql-over-domain-socket.patch}
+    sourceRoot = "source/client";
+    NODE_OPTIONS = "--openssl-legacy-provider";
+  };
 
-      # Workaround for https://github.com/svanderburg/node2nix/issues/275
-      sed -i -e "s|#!/usr/bin/env node|#! ${nodejs}/bin/node|" node_modules/node-gyp-build/bin.js
+  postBuild = ''
+    rm -rf client
+    cp -r ${clientDir} client
```

Comment:
Can we symlink this?

Reply:
It's safer not to, because code under `client/` references code in other directories. I only built it using a separate derivation to make it work with `buildNpmPackage`.